src/plugins/vis_augmenter/public/saved_augment_vis/utils/helpers.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get } from 'lodash';
+import { ISavedAugmentVis } from '../types';
+import {
+  PLUGIN_AUGMENTATION_ENABLE_SETTING,
+  PLUGIN_AUGMENTATION_MAX_OBJECTS_SETTING,
+} from '../../../common/constants';
+import { SavedAugmentVisLoader } from '../saved_augment_vis';
+import { getSavedAugmentVisLoader, getUISettings } from '../../services';
+import { IUiSettingsClient } from '../../../../../core/public';
+
+/**
+ * Create an augment vis saved object given an object that
+ * implements the ISavedAugmentVis interface
+ */
+export const createAugmentVisSavedObject = async (
+  AugmentVis: ISavedAugmentVis,
+  savedObjLoader?: SavedAugmentVisLoader,
+  uiSettings?: IUiSettingsClient
+): Promise<any> => {
+  // Using optional services provided, or the built-in services from this plugin
+  const loader = savedObjLoader !== undefined ? savedObjLoader : getSavedAugmentVisLoader();
```

Comment:
Wouldn't this be safer? What if `savedObjLoader` is `null`?
```suggestion
  const loader = savedObjLoader || getSavedAugmentVisLoader();
```